src/common/btree.c

```diff
@@ -2010,6 +2035,10 @@ btr_insert(struct btr_context *tcx, d_iov_t *key, d_iov_t *val, d_iov_t *val_out
 	union btr_rec_buf  rec_buf = {0};
 	int		   rc;
 
+	rc = btr_verify_key(tcx, key);
+	if (rc)
+		return rc;
+
```

Comment:
This looks redundant, it's already verified in btr_upsert().

It looks bit weird to me that the 'btr_verify_key()' calls are sporadically duplicated in both exported functions (dbtree_xxx) and internal functions (btr_xxx), I think the key should be verify in:
- Either all the exported dbtree_xxx() functions, or;
- The internal functions like btr_hkey_gen() and hkey_int_gen();
What do you think?

Reply:
Sure, let's move check earlier to dbtree_xxxx()